channeldb/graph.go

```diff
@@ -3347,6 +3347,37 @@ func (c *ChannelGraph) NewChannelEdgePolicy() *ChannelEdgePolicy {
 	return &ChannelEdgePolicy{db: c.db}
 }
 
+// MarkEdgeZombie attempts to mark a channel identified by its channel ID as a
+// zombie. This method is used on an ad-hoc basis, when channels need to be
+// marked as zombies outside the normal pruning cycle.
+func (c *ChannelGraph) MarkEdgeZombie(chanID uint64,
+	pubKey1, pubKey2 [33]byte) error {
+
+	err := kvdb.Batch(c.db, func(tx kvdb.RwTx) error {
+		edges := tx.ReadWriteBucket(edgeBucket)
+		if edges == nil {
+			return ErrGraphNoEdgesFound
+		}
+		zombieIndex := edges.NestedReadWriteBucket(zombieBucket)
+
+		// If there's no zombie index, then we couldn't have marked an
+		// edge as a zombie in the first place.
+		if zombieIndex == nil {
```

Comment:
I don't think this applies here? We should use `CreateIfNotExists` in case an edge has never been marked as a zombie before this.

Reply:
Fixed.